### PR TITLE
added some additional JSON schema features to the docs

### PR DESCRIPTION
**Pull request by amitb777**
# Description

added some additional JSON schema features to the docs

## Updated docs pages

- Blueprint (`/software-catalog/blueprint`)

### CODE REVIEW

File: docs/software-catalog/blueprint/blueprint.md [hunk 1]
```diff
@@ -414,6 +422,95 @@ print(config_prop["do_awesome_things"]) # prints: True
...
+
+```json showLineNumbers
+{
+  "type": "string",
+  "minLength": 2,
+  "maxLength": 3
+}
+```
+
+### Regular Expressions
```
Comment (MPTG94) [inline docs/software-catalog/blueprint/blueprint.md:437]: There is already a dedicated explanation about regex in the [docs](https://docs.getport.io/software-catalog/blueprint/#string-regular-expression-patterns). Please combine the new explanation with the old one, under the new **additional keywords** header

File: docs/software-catalog/blueprint/blueprint.md [hunk 2]
```diff
@@ -414,6 +422,95 @@ print(config_prop["do_awesome_things"]) # prints: True
...
+- _x_ < `exclusiveMaximum`
+
+```json showLineNumbers
+{
+  "type": "number",
+  "minimum": 0,
+  "exclusiveMaximum": 100
+}
+```
+
```
Comment (MPTG94) [inline docs/software-catalog/blueprint/blueprint.md:466]: Add a small explanation that the rule above allows x to be 0<=x<100

File: docs/software-catalog/blueprint/blueprint.md [hunk 3]
```diff
@@ -414,6 +422,95 @@ print(config_prop["do_awesome_things"]) # prints: True
...
+### Items
+
+List validation is useful for arrays of arbitrary length where each item matches the same schema. For this kind of array, set the items keyword to a single schema that will be used to validate all of the items in the array.
```
Comment (MPTG94) [inline docs/software-catalog/blueprint/blueprint.md:469]: There is already a section called [Special property types](https://docs.getport.io/software-catalog/blueprint/#special-property-types) which explains [object arrays](https://docs.getport.io/software-catalog/blueprint/#objects-array), [url arrays](https://docs.getport.io/software-catalog/blueprint/#urls-array) and [enums](https://docs.getport.io/software-catalog/blueprint/#enum), let's take those explanation and put them here, and delete the special property types header

Comment (MPTG94) [inline docs/software-catalog/blueprint/blueprint.md:469]: There is already a section called [Special property types](https://docs.getport.io/software-catalog/blueprint/#special-property-types) which explains [object arrays](https://docs.getport.io/software-catalog/blueprint/#objects-array), [url arrays](https://docs.getport.io/software-catalog/blueprint/#urls-array) and [enums](https://docs.getport.io/software-catalog/blueprint/#enum), let's take those explanation and put them here, and delete the special property types header

File: docs/software-catalog/blueprint/blueprint.md [hunk 4]
```diff
@@ -414,6 +422,95 @@ print(config_prop["do_awesome_things"]) # prints: True
...
+  "type": "array",
+  "items": {
+    "type": "number"
+  }
+}
+```
+
+### Properties
+
+The properties (key-value pairs) on an object are defined using the `properties` keyword. The value of `properties` is an object, where each key is the name of a property and each value is a schema used to validate that property. Any property that doesn’t match any of the property names in the `properties` keyword is ignored by this keyword.
```
Comment (MPTG94) [inline docs/software-catalog/blueprint/blueprint.md:484]:
```suggestion
It is possible to define the format and validation rules of key-value pairs inside an `object` property by using the `properties` key inside the `object` type definition.

The properties (key-value pairs) on an object are defined using the `properties` keyword. The value of `properties` is an object, where each key is the name of a property and each value is a schema used to validate that property. Any property that doesn’t match any of the property names in the `properties` keyword is ignored by this keyword.
```

File: docs/software-catalog/blueprint/blueprint.md [hunk 5]
```diff
@@ -414,6 +422,95 @@ print(config_prop["do_awesome_things"]) # prints: True
...
+  "type": "object",
+  "properties": {
+    "config_file_path": { "type": "string" },
+    "min_required_cpu": { "type": "number", "maximum": 8 }
+  }
+}
+```
+
+:::tip
+When using the `properties` keyword, you can also make use of the `additionalProperties` keyword to control the handling of extra stuff, that is, properties whose names are not listed in the properties keyword or match any of the regular expressions in the patternProperties keyword. By default, any additional properties are allowed.
```
Comment (MPTG94) [inline docs/software-catalog/blueprint/blueprint.md:499]:
```suggestion
When using the `properties` keyword, you can also make use of the `additionalProperties` keyword to control the handling of extra properties that were not explicitly defined. That is, properties whose names are not listed in the `properties` keyword of the `object` or match any of the regular expressions in the `patternProperties` keyword.

By default, any additional properties are allowed (`"additionalProperties": true`), by setting `additionalProperties` to `false`, only keys explicitly specified can be saved, and any other fields will be rejected (in the example below, only `config_file_path` and `min_required_cpu` will be allowed). 
```

Comment (MPTG94) [reply docs/software-catalog/blueprint/blueprint.md:499]: can you add an example where you use patternProperties here? It is written but not explained, so people will probably not know how to use it

**Review by MPTG94 — CHANGES_REQUESTED** [review]
Please apply fixes according to comments

File: docs/software-catalog/blueprint/blueprint.md (hunk 1 above)
Comment (amitb777) [reply docs/software-catalog/blueprint/blueprint.md:437]: done

File: docs/software-catalog/blueprint/blueprint.md (hunk 2 above)
Comment (amitb777) [reply docs/software-catalog/blueprint/blueprint.md:466]: done

File: docs/software-catalog/blueprint/blueprint.md (hunk 3 above)
Comment (amitb777) [reply docs/software-catalog/blueprint/blueprint.md:469]: done

Comment (amitb777) [reply docs/software-catalog/blueprint/blueprint.md:469]: done

File: docs/software-catalog/blueprint/blueprint.md (hunk 4 above)
Comment (amitb777) [reply docs/software-catalog/blueprint/blueprint.md:484]: done

File: docs/software-catalog/blueprint/blueprint.md (hunk 5 above)
Comment (amitb777) [reply docs/software-catalog/blueprint/blueprint.md:499]: I removed the part with the patternProperties, it's pretty advanced and I don't think anyone is going to use it...

**Review by MPTG94 — APPROVED** [review]
LGTM, please apply the requested fixes and merge

File: docs/software-catalog/blueprint/blueprint.md [hunk 6]
```diff
@@ -430,17 +438,100 @@ print(config_prop["do_awesome_things"]) # prints: True
...
+
+```json showLineNumbers
+{
+  "type": "string",
+  "minLength": 2,
+  "maxLength": 3
+}
+```
+
+### Regular Expressions
```
Comment (EllaFurman) [inline docs/software-catalog/blueprint/blueprint.md:481]: Expressions should be with a lower case "e"

File: docs/software-catalog/blueprint/blueprint.md [hunk 7]
```diff
@@ -478,30 +569,40 @@ The available special types and their usage examples are listed below:
 }
 ```
 
-### Enum
+### Properties
+
+It is possible to define the format and validation rules of key-value pairs inside an `object` property by using the `properties` keyword inside the `object` type definition.
```
Comment (EllaFurman) [inline docs/software-catalog/blueprint/blueprint.md:574]: it is = You can define...

**Pull request merged**